source/common/modules/markdown-editor/plugins/render-math.ts

```diff
@@ -0,0 +1,167 @@
+    // markdown, but comments shouldn't be included in rendering)
+    // Final check to avoid it for as long as possible, as getTokenAt takes
+    // considerable time.
+    let tokenTypeBegin = cm.getTokenTypeAt(myMarker.curFrom)
+    let tokenTypeEnd = cm.getTokenTypeAt(myMarker.curTo)
+    if ((tokenTypeBegin?.includes('comment')) || (tokenTypeEnd?.includes('comment'))) {
+      continue
+    }
+
+    let mathSpan = document.createElement('span')
+    mathSpan.classList.add('preview-math')
+
+    let textMarker = cm.getDoc().markText(
+      myMarker.curFrom, myMarker.curTo,
+      {
+        'className': 'test',
+        'clearOnEnter': true,
+        'replacedWith': mathSpan,
+        'inclusiveLeft': false,
+        'inclusiveRight': false
+      }
+    )
+
+    // Enable on-click closing of rendered Math elements.
+    mathSpan.onclick = (e) => { textMarker.clear() }
+
+    katex.render(myMarker.eq, mathSpan, { throwOnError: false, displayMode: myMarker.displayMode })
+
+    // Now the marker has obviously changed
+    textMarker.changed()
+  }
+}
+
+export class LineInfo {
+  constructor (
+    readonly lineNumber: number,
+    readonly text: string,
+    readonly modeName: string,
+    readonly tokenType?: string
+  ) {}
+}
+
+export class EquationMarker {
+  constructor (
+    readonly curFrom: CodeMirror.Position,
+    readonly curTo: CodeMirror.Position,
+    readonly eq: string,
+    readonly displayMode: boolean
+  ) {}
+}
+
+/**
+   * Finds all equations contained in a given string according to the Pandoc documentation
+   * on its tex_math_dollars-extension.
+   * More information: https://pandoc.org/MANUAL.html#math
+   * @param text the input string
+   * @param line the line number of the input
+   * @returns list of equations in the input
+   */
+export function findInlineEquations (text: string, line: number): EquationMarker[] {
+  let inlineMathRE = getInlineMathRenderRE(true) // Get the RE with the global flag set.
+  let newMarkers = []
+
+  let match
+  while ((match = inlineMathRE.exec(text)) !== null) {
+    if (match.groups === undefined) {
+      continue
+    }
+
+    newMarkers.push(new EquationMarker(
+      { 'ch': match.index, 'line': line },
+      { 'ch': match.index + match[0].length, 'line': line },
+      match.groups.eq ?? '',
+      // Equations surrounded by two dollars should be displayed as centred equation
+      (match.groups.dollar ?? '').length === 2
+    ))
+  }
+  return newMarkers
+}
+
+export function findEquations (lines: LineInfo[]): EquationMarker[] {
+  let insideMultiline = false // Are we inside a multiline math environment?
+  let eq: string[] = []
+  let fromLine = 0
+
+  // This array holds all markers to be inserted (either one in case of the
+  // final line of a multiline-equation or multiple in case of several
+  // inline equations).
+  let equations: EquationMarker[] = []
+
+  for (let line of lines) {
+    if (![ 'markdown', 'stex' ].includes(line.modeName)) continue
```

Comment:
While I personally did this myself for quite a long time, I have switched to always using curly brackets (just to make sure we don't overlook a continue, break, or something like this). Does make the code more verbose, but I have found it easier to read this way